applications/nrf5340_audio/src/audio_module/src/amod_api.h

```diff
@@ -0,0 +1,492 @@
+ *	                  (in milliseconds). Use K_NO_WAIT to return without waiting,
+ *	                  or K_FOREVER to wait as long as necessary.
+ *
+ * @return 0 if successful, error value
+ */
+int amod_data_rx(struct amod_handle *handle, struct aobj_block *block, k_timeout_t timeout);
+
+/**
+ * @brief Send and retrieve data from the module all data is consumed within
+ *        the call. The input data buffer maybe released once function retrurned.
+ *
+ * @param handle_tx   A handle to the module to send the input block to
+ * @param handle_rx   A handle to the module to receive an block from
+ * @param block_tx   Pointer to the audio data block to send
+ * @param block_rx   Pointer to the audio data block received
+ * @param timeout     Non-negative waiting period to wait for operation to complete
+ *	                  (in milliseconds). Use K_NO_WAIT to return without waiting,
+ *	                  or K_FOREVER to wait as long as necessary.
+ *
+ * @return 0 if successful, error value
+ */
+int amod_data_tx_rx(struct amod_handle handle_tx, struct amod_handle handle_rx,
+		    struct aobj_block *block_tx, struct aobj_block *block_rx, k_timeout_t timeout);
+
+/**
+ * @brief Helper function to configure the thread information for the module
+ *        set-up parameters structure.
+ *
+ * @param parameters   Pointer to the module set-up parameters
+ * @param description  Pointer to the modules private description
+ * @param stack        Memory block for the threads stack
+ * @param stack_size   Size of the threads stack
+ * @param priority     Priority of the thread insatnce, one of #amod_id
+ * @param msg_rx       Number of concurrent input messages
+ * @param msg_tx       Number of concurrent output messages
+ * @param data_slab    Slab to take a data item from
+ * @param data_size    Size of the data item that will be taken
+ *
+ * @return 0 if successful, error value
+ */
+int amod_parameters_configure(struct amod_parameters *parameters,
+			      struct amod_description *description, k_thread_stack_t *stack,
+			      size_t stack_size, int priority, struct data_fifo *msg_rx,
+			      struct data_fifo *msg_tx, struct k_mem_slab *data_slab,
+			      size_t data_size);
+
+/**
+ * @brief Helper function to return the base and instance names for a given
+ *        module handle.
+ *
+ * @param handle         A handle to this module instance
+ * @param base_name      A pointer to the name of the module
+ * @param instance_name  A pointer to the name of the current module instance
+ *
+ * @return 0 if successful, error value
+ */
+int amod_names_get(struct amod_handle *handle, char *base_name, char *instance_name);
+
+/**
+ * @brief Helper function to return the state of a given module handle.
+ *
+ * @param handle  A handle to this module instance
+ *
+ * @return 0 if successful, error value
+ */
+int amod_state_get(struct amod_handle *handle);
+
+/**
+ * @brief Helper function to attach a raw audio data buffer to an audio block.
```

Comment:
I think the term raw audio gives associations with PCM data, this could be LC3. If I've understood this correct all the parameters sent in here is written to the struct given by the pointer *block, right? Will this function just be something like:
```C
block->data_type = data_type;
block->data = data;
```
etc? If so, what is the benefit of this function?

Reply:
Yes I agree raw may confuse, so I will remove it and just leave it as "audio data".
Well it is to help the user bring a set of data items together without having to clutter their code. However the main reason is that if the naming changes or we pre-process the data before it is bound to the block or we add elements that are generated from the input, then the customer would have to change all their application code for this. So if we use the helper then it is all hidden and the customer need never know. This is a useful technique for protecting a customer's application from changes in modules.